### ISSUE TITLE
[JIT][PE] Profiling Executor should re-profile `DifferentiableGraph`

### ISSUE THREAD

**Issue by jjsjann123** [pytorch/pytorch#48094]
## 🚀 Feature

Profiling Executor should re-profile the forward graph of `DifferentiableGraph`

## Motivation

`differentiate` pass modifies the profiled graph into forward graph in `DifferentiableGraph`. This resulted in missing profiling information in the new graph. This is causing issue in codegen, which requested certain Tensor to be profiled and assumed access to profiling information.

e.g. in the code snippet below, `dropout` node is expanded to a group of nodes in `DifferentiableGraph` and no intermediate tensor in the expanded groups are getting profiled.

https://github.com/pytorch/pytorch/blob/49f0e5dfeb64a928c6a2368dd5f86573b07d20fb/torch/csrc/jit/runtime/symbolic_script.cpp#L1083-L1111

## Additional context

I have a toy PR that tries to use `GraphExecutor` instead of `Code` to run forward graph #47434. It's breaking the world since I blindly disabled `runDiffGraphPasses`, which I should probably only turn off the fusion pass there. But It could be more subtle than that and I might be poking the wrong thing.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
